### DISCUSSION TITLE
new config.yaml file

### DISCUSSION THREAD

**Discussion by yvr354** [open-edge-platform/anomalib#1736] [Q&A] [answered]
how can i get  a new formate of config.yaml file with all data and model and metrics and trainer parameters .

**Comment by samet-akcay** [comment]
Hi @yvr354, the CLI has `--print_config` flag can print out the full config to train a model on a certain dataset. For example, if you want to train a Padim model on MVTec dataset, you could do the following:

```bash
anomalib train --data anomalib.data.MVTec --model anomalib.models.Padim --print_config > config.yaml  
```

This will print the full config to the screen, from which you could save it as `config.yaml` file. You could then use this config to train like;

```bash
anomalib train -c config.yaml`
```

Let me know if this works.

**Comment by nguyenanhtuan1008** [comment]
@samet-akcay 
Sorry, How to use export model in new config? I would like to export it to Torch , openvino

**Comment by samet-akcay** [comment, marked as answer]
CLI Usage:
            1. To export as a torch ``.pt`` file you can run the following command.
                ```python
                anomalib export --model Padim --export_mode TORCH --data MVTec
                ```
            2. To export as an ONNX ``.onnx`` file you can run the following command.
                ```python
                anomalib export --model Padim --export_mode ONNX --data Visa --input_size "[256,256]"
                ```
            3. To export as an OpenVINO ``.xml`` and ``.bin`` file you can run the following command.
                ```python
                anomalib export --model Padim --export_mode OPENVINO --data Visa --input_size "[256,256]"
                ```
            4. You can also overrride OpenVINO model optimizer by adding the ``--mo_args.<key>`` arguments.
                ```python
                anomalib export --model Padim --export_mode OPENVINO --data Visa --input_size "[256,256]" \
                    --mo_args.compress_to_fp16 False
                ```

**Reply by nguyenanhtuan1008** [reply]
@samet-akcay 
Thank you so much.

**Reply by samet-akcay** [reply]
Sure, for more details, you could refer to the following documentation
https://anomalib.readthedocs.io/en/latest/markdown/guides/reference/engine/index.html#anomalib.engine.Engine.export
